packages/core/src/utils/traceData.ts

```diff
@@ -0,0 +1,87 @@
+import type { Client, Scope, Span } from '@sentry/types';
+import {
+  TRACEPARENT_REGEXP,
+  dynamicSamplingContextToSentryBaggageHeader,
+  generateSentryTraceHeader,
+  logger,
+} from '@sentry/utils';
+import { getClient, getCurrentScope } from '../currentScopes';
+import { getDynamicSamplingContextFromClient, getDynamicSamplingContextFromSpan } from '../tracing';
+import { getActiveSpan, getRootSpan, spanToTraceHeader } from './spanUtils';
+
+/**
+ * Extracts trace propagation data from the current span or from the client's scope (via transaction or propagation
+ * context) and serializes it to `sentry-trace` and `baggage` values to strings. These values can be used to propagate
+ * a trace via our tracing Http headers or Html `<meta>` tags.
+ *
+ * This function also applies some validation to the generated sentry-trace and baggage values to ensure that
+ * only valid strings are returned.
+ *
+ * @param span a span to take the trace data from. By default, the currently active span is used.
+ * @param scope the scope to take trace data from By default, the active current scope is used.
+ * @param client the SDK's client to take trace data from. By default, the current client is used.
+ *
+ * @returns an object with the tracing data values. The object keys are the name of the tracing key to be used as header
+ * or meta tag name.
+ */
+export function getTraceData(
+  span?: Span | undefined,
+  scope?: Scope,
+  client?: Client,
+): { 'sentry-trace': string; baggage?: string } {
```

Comment:
l: Could we expose this as type?

Reply:
Do we need the type? I don't think we should export things we don't explicitly need

Reply:
Fair

Reply:
I extracted it to a type but didn't export it for now